ui/uppy/src/components/PickerPanelTopBar.jsx

```diff
@@ -76,7 +76,30 @@ function UploadStatus({
 }
 
 function PanelTopBar(props) {
-  const { i18n, isAllComplete, hideCancelButton, maxNumberOfFiles, toggleAddFilesPanel, uppy } = props;
+  const {
+    i18n,
+    isAllComplete,
+    isAllErrored,
+    isAllPaused,
+    files,
+    invalidFiles,
+    newFiles,
+    hideCancelButton,
+    maxNumberOfFiles,
+    toggleAddFilesPanel,
+    uppy
+  } = props;
+  const uploadingState = getUploadingState(isAllErrored, isAllComplete, isAllPaused, files);
+  const invalidFilesCount = Object.values(invalidFiles).filter((invalid) => invalid).length;
+  const newFilesCount = newFiles.length;
+  const disableProceed =
+    // if there are active uploads
+    uploadingState === 'uploading' ||
+    // there are no new files to upload (all files already uploaded)
+    !newFilesCount ||
+    // if there are invalid files (due to site policies) waiting to be uploaded
+    (invalidFilesCount && uploadingState === 'waiting');
```

Comment:
- This looks like if files are added whilst other files are being uploaded, the proceed will be disabled?
- If newFilesCount is a number, let's express that condition as dealing with numbers (i.e. newFilesCount === 0)

Reply:
Updated, it'll go like this:
With `autoproceed` false, if you add files and hit the `Start Uploads` button, it'll start uploading the added files, and `Start Uploads` will be disabled.
If while uploading, more files are added, then the `Start Uploads` button will be re-enabled, and if the button is clicked, it'll start uploading only the newly added files (the previous uploads continue).

disableProceed validation became simpler because now `newFiles` doesn't include the files pending due to policies validation (already started to upload), so by only validating the newFiles count is enough.

Also, I missed the individual proceed comment, it is now added, and the proceed button it'll only show at the beginning (when autoProceed is false), and the batch proceed button works as expected (not disabled unless there are no pending uploads)